### PR TITLE
fix(cli): add otel configuration

### PR DESCRIPTION
**Pull request by samnguyen-outreach**
## What this PR does / why we need it

- Fixes panic in bootstrap CLI programs after introduction of Otel code:

```
panic: It looks like you upgraded gobox without upgrading bootstrap. To correct the issue:
  - Update bootstrap to v10.2.0-rc.3 or later
  - add 'tracing: opentelemetry' to your service.yaml

github.com/getoutreach/gobox/pkg/trace.InitTracer({0x104b3d960, 0x1400012a800}, {0x104275c36, 0xc})
	/Users/sam.nguyen/.asdf/installs/golang/1.17.9/packages/pkg/mod/github.com/getoutreach/gobox@v1.43.0/pkg/trace/trace.go:134 +0xbc
github.com/getoutreach/gobox/pkg/cli.setupTracer({0x104b3d960, 0x1400012a800}, {0x104275c36, 0xc})
	/Users/sam.nguyen/.asdf/installs/golang/1.17.9/packages/pkg/mod/github.com/getoutreach/gobox@v1.43.0/pkg/cli/bootstrap.go:136 +0x40
```

## Jira ID

[QSS-1176]

## Notes for your reviewers

- I'm not sure this is correct, but it does fix the panic in my application.

[QSS-1176]: https://outreach-io.atlassian.net/browse/QSS-1176?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: pkg/cli/bootstrap.go [hunk 1]
```diff
@@ -72,6 +72,15 @@ func overrideConfigLoaders(honeycombAPIKey, dataset string, tracingDebug bool) {
 					SamplePercent: 100,
 				},
 			}
+			traceConfig.Otel = trace.Otel{
+				Enabled:       traceConfig.Honeycomb.Enabled,
+				Endpoint:      traceConfig.Honeycomb.APIHost,
```
Comment (artemchernyak-outreach) [inline pkg/cli/bootstrap.go:77]: You may need to manually set this one to `api.honeycomb.io`. Otel will fail to send traces if you provide the scheme.

Comment (artemchernyak-outreach) [inline pkg/cli/bootstrap.go:77]: You may need to manually set this one to `api.honeycomb.io`. Otel automatically provides the scheme and will fail if it's also provided as part of the endpoint.

Comment (artemchernyak-outreach) [inline pkg/cli/bootstrap.go:77]: You will need to set this to `api.honeycomb.io` manually. Otel automatically provides the scheme and will fail if it's also provided as part of the endpoint config.

**Review by artemchernyak-outreach — CHANGES_REQUESTED** [review]

**Comment by artemchernyak-outreach** [comment]
Thanks for providing the update for this! I think this is the right approach to fix this issue. Please let me know if you run into any other issues once this change gets applied.

Sorry for the miss in the original implementation. I did not realize that the config was provided differently for cli apps.

File: pkg/cli/bootstrap.go (hunk 1 above)
Comment (samnguyen-outreach) [reply pkg/cli/bootstrap.go:77]: Thanks @artemchernyak-outreach .  Do you mean just like this? 
```suggestion
				Endpoint:      "api.honeycomb.io",
```

Comment (artemchernyak-outreach) [reply pkg/cli/bootstrap.go:77]: Yup, exactly like that. I probably should have have just done a change suggestion on that one.

**Review by artemchernyak-outreach — APPROVED** [review]

**Comment by samnguyen-outreach** [comment]
Thanks @artemchernyak-outreach!  I've added a test to cover the regression. I'll let you merge as you see fit, as i'm not familiar with the release process for this code.

File: pkg/trace/roundtripper_test.go [hunk 2]
```diff
@@ -509,55 +509,3 @@ func (suite) initHeaderRoundTripperState(t *testing.T) *roundtripperState {
 
 	return &roundtripperState{recorder, server}
 }
-
-func (suite) TestRoundripNoTracer(t *testing.T) {
```
Comment (artemchernyak-outreach) [inline pkg/trace/roundtripper_test.go:513]: We need to completely remove this test because now we panic when there is no tracer in the config.

**Comment by samnguyen-outreach** [comment]
🟢  Tested against `polaroid-cli` with no panic:

```
❯ go get -v github.com/getoutreach/gobox@fix/samnguyen-outreach/cli-otel
go: downloading github.com/getoutreach/gobox v1.43.2-0.20220711144700-8afbd68885f5
github.com/getoutreach/gobox
go get: upgraded github.com/getoutreach/gobox v1.43.2-0.20220709000610-5b04f734ac23 => v1.43.2-0.20220711144700-8afbd68885f5

❯ make
❯ bin/polaroid-cli -v
polaroid-cli version v1.14.15
```

**Pull request merged**